src/api/controller/api/annotation.validator.js

```diff
@@ -0,0 +1,66 @@
+import { default as z } from 'zod';
+
+export const annotationSchema = z.object({
+    resourceId: z.string({
+        required_error: 'error_required',
+    }),
+    fieldId: z.string().nullish().default(null),
```

Comment:
Didn't we say we would use a path instead, to support thing like pointing to a value inside an array of values ?